distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Ordering/Natural_Order.enso

```diff
@@ -0,0 +1,69 @@
+from Standard.Base import all
+
+import Standard.Base.Data.Text.Regex
+import Standard.Base.Data.Text.Regex.Mode
+import Standard.Base.Data.Ordering.Vector_Lexicographic_Order
+polyglot java import org.enso.base.Regex_Utils
+
+## Compares two text values according to the natural dictionary ordering.
+
+   > Example
+     Compare two texts according to the natural dictionary ordering.
+
+         Natural_Order.compare "a2" "a100" == Ordering.Less
+
+   > Example
+     Sort a vector of texts according to the natural dictionary ordering.
+
+         ["a2", "a1", "a100", "a001", "a0001"].sort by=Natural_Order.compare . should_equal ["a0001", "a001", "a1", "a2", "a100"]
+compare : Text -> Text -> Ordering
+compare text1 text2 =
+    nat1 = here.to_natural_key text1
+    nat2 = here.to_natural_key text2
+    ## TODO [RW] for some reason removing this `res` and returning the result
+       directly will make Vector.sort behave badly. This needs to be investigated
+       and filed as a bug, then this comment can be updated with a reference to a
+       relevant ticket.
```

Comment:
Such things should be done before submitting code to the codebase. Otherwise this code would never be updated. If there is a problem, it should be investigated, a bug should be filled and linked here.